include/deal.II/base/bounding_box.h

```diff
@@ -0,0 +1,108 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 1998 - 2016 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_base_bounding_box_h
+#define dealii_base_bounding_box_h
+
+
+#include <deal.II/base/config.h>
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/point.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * A class that represents a bounding box in a space with arbitrary dimension
+ * <tt>spacedim</tt>.
+ *
+ * Objects of this class are used to represent bounding boxes. They are,
+ * among other uses, useful in parallel distributed meshes to give a general
+ * description the owners of each portion of the mesh.
+ *
+ * Boundig boxes are represented by two vertices and, geometrically, are
+ * represented by the following:
+ * 1 d: a segment (represented by its vertices)
+ * 2 d: a rectangle (represented by two vertices V opposed to the center)
```

Comment:
can user provide any pair or the one you describe here? 
i.e. bottom left + top right vs
bottom right + top left.

In dealii there is already a function https://www.dealii.org/developer/doxygen/deal.II/namespaceGridTools.html#a0374dd9eae73c3c2badaa479e8ba8d55
which defines it as bottom left + top right. So this can should be supported for consistency.

Reply:
I shall use the bottom left, top right convention then. This means I can also modify the is inside function.

Reply:
yes, and then it's also easy to check if the points satisfy bottom-left-to-right.